### PR TITLE
Sync refactor

### PR DESCRIPTION
**Pull request by ewels**
Epic refactor / rewrite of the pipeline template synchronisation code.

Concepts and some code taken from @sven1103, rewritten to be part of the main `nf-core` command. Now works with personal forks, and should be easier to test and reuse.

### CODE REVIEW

**Comment by ewels** [comment]
Retriggered tests after upstream fix in #424 🤞

**Comment by ewels** [comment]
Ok, if you all ask then I will capitulate 🙇 😅 

Quickly refactored before bed - untested though.

File: .travis.yml [hunk 1]
```diff
@@ -57,4 +57,4 @@ deploy:
 
 # Sync pipelines with possible nf-core template changes
 after_deploy:
-  - ./bin/sync
+  - nf-core sync-all
```
Comment (MaxUlysse) [inline .travis.yml]:
```suggestion
  - nf-core sync --all
```

File: README.md [hunk 2]
```diff
@@ -16,6 +16,8 @@ A python package with helper tools for the nf-core community.
 * [`nf-core create` - Create a new workflow from the nf-core template](#creating-a-new-workflow)
 * [`nf-core lint` - Check pipeline code against nf-core guidelines](#linting-a-workflow)
 * [`nf-core bump-version` - Update nf-core pipeline version number](#bumping-a-pipeline-version-number)
+* [`nf-core sync` - Synchronise a pipeline TEMPLATE branch](#sync-a-pipeline-with-the-template)
+* [`nf-core sync-all` - Sync all nf-core pipelines and trigger PRs](#sync-all-nf-core-pipelines)
```
Comment (MaxUlysse) [inline README.md]:
```suggestion
* [`nf-core sync --all` - Sync all nf-core pipelines and trigger PRs](#sync-all-nf-core-pipelines)
```

Comment (MaxUlysse) [reply README.md]: Or do we need to remove this line?

**Comment by ewels** [comment]
Good spot, fixed - thanks @MaxUlysse!

**Comment by ewels** [comment]
Don't merge yet - `nf-core sync --all` is buggy:

```console
$ nf-core sync --all

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

Usage: nf-core sync [OPTIONS] <pipeline directory>
Try "nf-core sync --help" for help.

Error: Missing argument "<pipeline directory>".
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 ### Tools helper code
 
-* Updated Blacklist of synced pipelines
```
Comment (MaxUlysse) [inline CHANGELOG.md]: Removing stuff from the CHANGELOG?

Comment (ewels) [reply CHANGELOG.md]: Yes, it doesn’t make sense to say in the changelog for this release that we have updated the blacklist, when a couple of lines later say that it’s removed..

**Comment by ewels** [comment]
Should now be fixed I think 👍

**Pull request merged**